controllers/glance_controller.go

```diff
@@ -866,6 +866,13 @@ func (r *GlanceReconciler) apiDeploymentCreateOrUpdate(
 	if instance.Spec.KeystoneEndpoint == apiName {
 		apiAnnotations[glance.KeystoneEndpoint] = "true"
 	}
+
```

Comment:
can the topology of a deployment be updated after it get created, or is it immutable? We should probably watch the topology, like we do with the secrets. if it can be updated, all good, if not we have to think about what we should do in that case.

Reply:
It's not immutable and I'm able to see reconciliation loops being triggered by a change in the topology.
Adding the same watching logic here would  simply be redundant, because `glanceapi_controller` (where I pass the topology reference) has already such watcher, and it triggers a reconciliation if something changes even when the `topologyRef` is a glance top level parameter (tested locally).

Reply:
ah right, have not looked closely at the api controller